monitoring/monitoring.go

```diff
@@ -186,28 +219,23 @@ func NewGauge(name, description string, opts ...Options) Metric {
 // NewDerivedGauge creates a new Metric with an aggregation type of LastValue that generates the value
 // dynamically according to the provided function. This can be used for values based on querying some
 // state within a system (when event-driven recording is not appropriate).
-func NewDerivedGauge(name, description string, valueFn func() float64) DerivedMetric {
-	return NewDerivedGaugeWithLabels(name, description, nil, valueFn)
-}
-
-// NewDerivedGaugeWithLabels creates a new Metric that takes its value from the provided function with a set of const labels.
-func NewDerivedGaugeWithLabels(name, description string, labels map[string]string, valueFn func() float64) DerivedMetric {
-	constLabels := make(map[metricdata.LabelKey]metricdata.LabelValue)
-	for k, v := range labels {
-		constLabels[metricdata.LabelKey{Key: k}] = metricdata.NewLabelValue(v)
-	}
+func NewDerivedGauge(name, description string, opts ...DerivedOptions) DerivedMetric {
```

Comment:
Overall, this work.
```console
istio_agent_cert_expiry_seconds{resource_name="default"} 86354.308602142
istio_agent_cert_expiry_seconds{resource_name="default1"} 86354.308604592
```

BUT, this changes will broken all old implements.
If there's an automator, that will failed on here 
https://github.com/istio/istio/blob/b020516b7ddcfc19f4b635a3c85a8eb7b25b2b12/pilot/pkg/bootstrap/monitoring.go#L44

Reply:
That is ok, we do not need backwards compatibility in this repo

Reply:
It will require a manual update of the dep for istio/istio. I don't think the `DerivedGauge` stuff has been used anywhere other than that single location, so it should be easy to update hopefully.

Reply:
I will update istio/istio once this is merged.